diff --git c/libshims/Android.bp i/libshims/Android.bp
index 44bc668fbe11fa096bc0d4d03fa9299ce386c252..e988ef4082df09023248990939111e8c443c11b0 100644
--- c/libshims/Android.bp
+++ i/libshims/Android.bp
@@ -113,6 +113,8 @@ cc_library_shared {
 
 cc_library_shared {
     name: "libui_shim",
+    vendor: true,
+    compile_multilib: "32",
     srcs: [
         "libui/GraphicBuffer.cpp",
         "libui/GraphicBufferMapper.cpp",
diff --git c/libshims/libui/GraphicBuffer.cpp i/libshims/libui/GraphicBuffer.cpp
index bd40220cf21d06c41eb315b48a569a3d0905327c..2e5435fd95505dd74b471a19942ea6e1162c8d3a 100644
--- c/libshims/libui/GraphicBuffer.cpp
+++ i/libshims/libui/GraphicBuffer.cpp
@@ -8,36 +8,32 @@
 using android::PixelFormat;
 using android::status_t;
 
-extern "C" void *
+extern "C" void
 _ZN7android13GraphicBufferC1EjjijNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE(
     void *thisptr, uint32_t inWidth, uint32_t inHeight, PixelFormat inFormat,
     uint32_t inUsage, std::string requestorName = "<Unknown>");
 
-extern "C" void *_ZN7android13GraphicBufferC1Ejjij(void *thisptr,
-                                                   uint32_t inWidth,
-                                                   uint32_t inHeight,
-                                                   PixelFormat inFormat,
-                                                   uint32_t inUsage) {
-  return _ZN7android13GraphicBufferC1EjjijNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE(
+extern "C" void _ZN7android13GraphicBufferC1Ejjij(void *thisptr,
+                                                  uint32_t inWidth,
+                                                  uint32_t inHeight,
+                                                  PixelFormat inFormat,
+                                                  uint32_t inUsage) {
+  _ZN7android13GraphicBufferC1EjjijNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE(
       thisptr, inWidth, inHeight, inFormat, inUsage);
 }
 
-extern "C" void *_ZN7android13GraphicBufferC1EjjijjjP13native_handleb(
-    void *thisptr, const native_handle_t *handle,
-    android::GraphicBuffer::HandleWrapMethod method, uint32_t width,
-    uint32_t height, int format, uint32_t layerCount, uint64_t usage,
-    uint32_t stride);
+extern "C" void _ZN7android13GraphicBufferC1EjjijjjP13native_handleb(
+    void *thisptr, uint32_t inWidth, uint32_t inHeight, int inFormat,
+    uint32_t inLayerCount, uint32_t inUsage, uint32_t inStride,
+    native_handle_t *inHandle, bool keepOwnership);
 
-extern "C" void *_ZN7android13GraphicBufferC1EjjijjP13native_handleb(
+extern "C" void _ZN7android13GraphicBufferC1EjjijjP13native_handleb(
     void *thisptr, uint32_t inWidth, uint32_t inHeight, int inFormat,
     uint32_t inUsage, uint32_t inStride, native_handle_t *inHandle,
     bool keepOwnership) {
-  android::GraphicBuffer::HandleWrapMethod inMethod =
-      (keepOwnership ? android::GraphicBuffer::TAKE_HANDLE
-                     : android::GraphicBuffer::WRAP_HANDLE);
-  return _ZN7android13GraphicBufferC1EjjijjjP13native_handleb(
-      thisptr, inHandle, inMethod, inWidth, inHeight, inFormat,
-      static_cast<uint32_t>(1), static_cast<uint64_t>(inUsage), inStride);
+  _ZN7android13GraphicBufferC1EjjijjjP13native_handleb(
+      thisptr, inWidth, inHeight, inFormat, 1, inUsage, inStride, inHandle,
+      keepOwnership);
 }
 
 extern "C" status_t _ZN7android13GraphicBuffer4lockEjPPvPiS3_(
